packages/vite/src/node/config.ts

```diff
@@ -383,9 +383,20 @@ export async function resolveConfig(
   }
 
   // resolve root
-  const resolvedRoot = normalizePath(
-    config.root ? path.resolve(config.root) : process.cwd()
-  )
+  let resolvedRoot: string
+  if (!config.root) {
+    resolvedRoot = process.cwd()
```

Comment:
So the suggested behavior is:
1. If there isn't a config file, use `process.cwd()`
2. else, the root is relative to its dirname and if not included, the root is `'./'` (so the root is the config dirname)

I think this isn't as surprising as the PR but makes working with config files inside of the project a lot more complex. It is also strange to me that the presence of a vite config file will affect things so much. For example, I would expect that if I have a `vite.config.js` and it ends up empty and I remove it, things should change. So I'm starting to see why the current implementation may have ended up using `cwd()`.

Maybe the easiest way is to leave code as is, and document that everything is relative to `process.cwd()`. I don't think that having to type `cd dir && vite` is that bad anyways.

Reply:
I noticed the cli was what I felt tricky. The preference of `root` argument and `config.root` is mixed. It is similar to what you pointed out: "I would expect that if I have a vite.config.js and it ends up empty and I remove it, things should change. ".
Maybe `inlineRoot` always needs to be resolved from `process.cwd()`.

## `root` value

### config file not exists
| command | current | this PR | suggested |
|------------|----------|--------|-------------|
|`vite`|`.`|`.`|`.`|
|`vite .`|`.`|`.`|`.`|
|`vite  ./bar`|`./bar`|`./bar`|`./bar`|

### `./foo/vite.config.js` does not have `root` defined
| command | current | this PR | suggested |
|------------|----------|--------|-------------|
|`vite -c ./foo/vite.config.js`|`.`|`.`|`./foo`|
|`vite -c ./foo/vite.config.js .`|`.`|`./foo`|`./foo`|
|`vite -c ./foo/vite.config.js ./bar`|`./bar`|`./foo/bar`|`./foo/bar`|

### `./foo/vite.config.js` have `root='./baz'` defined
| command | current | this PR | suggested |
|------------|----------|--------|-------------|
|`vite -c ./foo/vite.config.js`|`./baz`|`./foo/baz`|`./foo/baz`|
|`vite -c ./foo/vite.config.js .`|`.`|`./foo`|`./foo`|
|`vite -c ./foo/vite.config.js ./bar`|`./bar`|`./foo/bar`|`./foo/bar`|

Reply:
Just in case, I wanted to write: "I would expect that if I have a vite.config.js and it ends up empty and I remove it, things **shouldn't** change."

What you suggest is consistent and it would be fine. What I don't know if it is the best given that we are going to break a lot of config files (all the people placing their configs in a nested dir) and I don't know if it really ends up being more consistent that saying that root is resolved against `cwd()` and force people to call vite at the right place or pass an inline root

Maybe something like:

## `root` value

### config file not exists
| command | current | this PR | suggested | alternative |
|------------|----------|--------|-------------|----------|
|`vite`|`.`|`.`|`.`|`.`|
|`vite .`|`.`|`.`|`.`|`.`|
|`vite  ./bar`|`./bar`|`./bar`|`./bar`|`./bar`|

### `./foo/vite.config.js` does not have `root` defined
| command | current | this PR | suggested | alternative |
|------------|----------|--------|-------------|----------|
|`vite -c ./foo/vite.config.js`|`.`|`.`|`./foo`|`.`
|`vite -c ./foo/vite.config.js .`|`.`|`./foo`|`./foo`|`.`
|`vite -c ./foo/vite.config.js ./bar`|`./bar`|`./foo/bar`|`./foo/bar`|`/.bar`

### `./foo/vite.config.js` have `root='./baz'` defined
| command | current | this PR | suggested | alternative |
|------------|----------|--------|-------------|----------|
|`vite -c ./foo/vite.config.js`|`./baz`|`./foo/baz`|`./foo/baz`|`.baz`|
|`vite -c ./foo/vite.config.js .`|`.`|`./foo`|`./foo`|`.baz`|
|`vite -c ./foo/vite.config.js ./bar`|`./bar`|`./foo/bar`|`./foo/bar`|`.baz`|

So the base root is `cwd()`, and root is relative to it.